sys/kern/interrupt.c

```diff
@@ -60,14 +64,27 @@ static void insert_handler(intr_event_t *ie, intr_handler_t *ih) {
   ie->ie_count++;
 }
 
-void intr_event_add_handler(intr_event_t *ie, intr_handler_t *ih) {
+static void ie_add_handler(intr_event_t *ie, intr_handler_t *ih) {
   WITH_SPIN_LOCK (&ie->ie_lock) {
-    insert_handler(ie, ih);
+    ie_insert_handler(ie, ih);
     if (ie->ie_count == 1 && ie->ie_enable)
       ie->ie_enable(ie);
   }
 }
 
+intr_handler_t *intr_event_add_handler(intr_event_t *ie, ih_filter_t *filter,
+                                       ih_service_t *service, void *arg,
+                                       const char *name) {
+  intr_handler_t *ih = kmalloc(M_INTR, sizeof(intr_handler_t), M_ZERO);
+  ih->ih_filter = filter;
+  ih->ih_service = service;
+  ih->ih_argument = arg;
+  ih->ih_name = name;
+  ih->ih_prio = 0; /* ? */
```

Comment:
Maybe an additional argument?

Reply:
I'm having serious doubts about usefulness of `ih_prio`. I marked this field in `intr_handler` as good candidate for removal.